cmd/ingest/ingest.go

```diff
@@ -201,27 +201,41 @@ func (ing *ingester) transferEntity(e *entityInfo) {
 		ing.errorf("failed to get information from destination charmstore on %q: %v", e.id, err)
 		return
 	}
-	// The entity doesn't exist in the destination, so copy it.
-	r, err := ing.src.getArchive(e.id)
-	if err != nil {
-		ing.errorf("cannot get archive for %v: %v", e.id, err)
-		return
+
+	sr := &seekReopener{
+		open: func() (io.ReadCloser, error) {
+			// The entity doesn't exist in the destination, so copy it.
```

Comment:
This comment doesn't seem to make sense here.

Reply:
moved